enhancements/E-0001-secret-handling.md

```diff
@@ -0,0 +1,748 @@
+  src:
+    namespace: kube-system
+    name: bootstrap-token-abcdef
+  dest:
+  - namespace: default
+    name: synced-bootstrap-token
+```
+
+Example that polls the input secret:
+```yaml
+apiVersion: package-operator.run/v1alpha1
+kind: SecretSync
+metadata:
+  name: bootstrap-token
+spec:
+  paused: false
+  strategy:
+    poll:
+      interval: 15m
+  src:
+    namespace: kube-system
+    name: bootstrap-token-abcdef
+  dest:
+  - namespace: default
+    name: synced-bootstrap-token
+```
+
+The api must:
+- Default `.spec.strategy` to `{watch: {}}` (default to the watch strategy if unspecified).
+- Enforce that the `.spec.strategy` is immutable and cannot be chanded on a live SecretSync object to avoid cache/finalizer confusions.
+- Enforce that only `.spec.stategy.watch` or `.spec.stategy.poll` are set and not both.
+- Enforce that, if the `poll` strategy is in use, then `.spec.stategy.poll.interval` must be set.
+- Enforce a lower bound on `.spec.stategy.poll.interval` to prevent DoSing the pko-manager.
+- Enforce an upper bound on the length of `.spec.dest[]` which must not exceed 32 items to prevent DoSing the pko-manager.
+- Enforce unique items in `.spec.dest[]`.
+- Report a Paused status condition.
+- Report a Sync status condition.
+  - (Optional - feasibility unverified) PKO probes reflecting degraded workload health when a synced secret vanishes or during out-of-sync times after a source secret has been changed and the resynchronization hasn't happened yet
+- Report a Phase, like our other APIs do, too.
+
+#### Package controller changes for SecretSync support features
+
+TODO: run this by [@thetechnick](https://github.com/thetechnick) again. Should we provide this feature and if we do, should we build an opt out mechanism? I think yes because we don't want to unneccessarily restrict the freedom of what can be in a package.
+
+The package controller should be changed to include the following "synergy" feature for SecretSync objects in Packages.
+
+To prevent unforeseen mishaps, when the rendered output of a package includes a SecretSync Object, PKO must validate that the object reference in `SecretSync.spec.src` points to a secret that is aliased in `.spec.secretRefs`.
+This cannot be relied on as a security feature because it is very easy to include wrapper objects (like OpenShift's Template objects) that act as trojan horses, but it serves as a guiding rail that steers package maintainers away from including hard-coded SecretSyncs in their packages and instead honor the user by declaring and aliasing secret references in the PackageManifest, so that they can be supplied dynamically at deployment time.
+
+#### Adapt kubectl-package
+
+Kubectl-package must introduce a new `inspect` command that prints its specified secret requirements.
+The output format is not specified here but it should be open to being expanded with other data about the package and being json/yaml formattable to aid further feature work in the future.
```

Comment:
Please create a new concept to outline the inspect command.
While the feature introduced by this proposal makes an inspect command useful, I would not piggy back more features into this concept.